formatter/formatter.go

```diff
@@ -49,3 +64,28 @@ func (f *TextFormatter) parseLevel(level logrus.Level) string {
 
 	return f.levelDesc[level]
 }
+
+// Format renders a single log entry
+func (f *SyslogFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	var fields string
+	keys := make([]string, 0, len(entry.Data))
+	for k, v := range entry.Data {
+		if k == "source" {
+			continue
+		}
+		keys = append(keys, fmt.Sprintf("%s: %v", k, v))
+	}
+
+	if len(keys) > 0 {
+		fields = fmt.Sprintf("[%s] ", strings.Join(keys, ", "))
+	}
+	return []byte(fmt.Sprintf("%s%s\n", fields, entry.Message)), nil
+}
+
+func (f *SyslogFormatter) parseLevel(level logrus.Level) string {
```

Comment:
DRY violation ;) this one is identical to https://github.com/netbirdio/netbird/pull/2292/files#diff-8ed12580b93d2c4e1114f9ca9807e9be22aa46d50e2f47e8cc912e1ee1bfa547R40 but I don't know how to make this function generic

we can merge it as is or you can give me some hints